geopandas/sindex.py

```diff
@@ -252,11 +252,16 @@ def intersection(self, coordinates):
 
         @property
         def size(self):
-            return len(self.leaves()[0][1])
+            if hasattr(self, "_size"):
+                size = self._size
+            else:
+                size = len(self.leaves()[0][1])
```

Comment:
It looks like rtree index always returns a list of tuples, even if empty:
```
>>> RTreeIndex(np.array([])).leaves()[0][1]
[]
```

It might be good to add a comment indicating that an item is always present at that index(`[0][1]`), even if rtree spatial index is empty, so as to avoid future confusion that we might be checking something that may or may not be present.

Reply:
Done.